test/runtests.jl

```diff
@@ -141,3 +141,15 @@ end
 
 end
 
+@testset "Anonymous function constructors" begin
+    function multiplyer(a, b)
+        x -> x * a * b
+    end
+
+    mult11 = multiplyer(1, 1)
+    @test mult11(1) === 1
+    mult23 = ConstructionBase.constructorof(typeof(mult11))(2.0, 3.0)
+    @test mult23(1) === 6.0
+    multbc = ConstructionBase.constructorof(typeof(mult23))("b", "c")
+    @test multbc("a") == "abc" 
+end
```

Comment:
Another good test case would be a used defined struct
```julia
struct Adder{V} <: Function
    value::V
end
(o::Adder)(x) = o.value + x
...
```

Reply:
Right I didn't even think of that. I guess it can break some objects that would currently work, like this:

```julia
struct Adder{V} <: Function
    value::V
    int::Int
end
```

Maybe instead of the allowing type error that currently occurs we should check if the number of args matches the number of parameters in `F`, and if they don't fall back to regular `constructorof`. Then the above will still work.

Reply:
Urgs, thats a good point. I agree with your fallback approach.

Reply:
We also need to check that `T.parameters == T.types` or this will break too:

```julia
struct AddTuple{T} <: Function
    tuple::Tuple{T,T,T}
end
```

Reply:
Ok these are covered if you want to have another look at it